bitmex-stream/src/lib.rs

```diff
@@ -16,16 +25,41 @@ pub use tokio_tungstenite::tungstenite::Error;
 pub fn subscribe<const N: usize>(
     topics: [String; N],
     network: Network,
+    credentials: Option<Credentials>,
 ) -> impl Stream<Item = Result<String, Error>> + Unpin {
+    let url = network.to_url();
+    let url = format!("wss://{url}/realtime");
+
     let stream = stream! {
         tracing::debug!("Connecting to BitMex realtime API");
 
-        let subscription = topics.join(",");
-        let url = network.to_url();
-        let (mut connection, _) = tokio_tungstenite::connect_async(format!("wss://{url}/realtime?subscribe={subscription}")).await?;
+        let (mut connection, _) = tokio_tungstenite::connect_async(url.clone())
+            .await.context("Could not connect to websocket")?;
 
         tracing::info!("Connected to BitMex realtime API");
 
+        let mut messages = vec![];
+        if let Some(credentials) = credentials {
+            let start = SystemTime::now();
+            let expires = start
+                .duration_since(UNIX_EPOCH)
+                .expect("Time went backwards")
+                .add(Duration::from_secs(5))
+                .as_secs();
+            let signature = credentials.sign(Method::GET, expires, &Url::parse(url.as_str())?, "");
+            messages.push(tungstenite::Message::try_from(signature.into_command())?);
+        }
+
+        messages.push(tungstenite::Message::try_from(Command::Subscribe(
```

Comment:
What's the advantage of sending multiple subscribe messages? 
Did you make sure that subscribing twice does not overwrite the previous subscription?

Reply:
> What's the advantage of sending multiple subscribe messages?

We have the option to subscribe to multiple topics. 

>  Did you make sure that subscribing twice does not overwrite the previous subscription?

That's the way how it works :) Yes, I've tried. 
Maybe it's possible to subscribe to multiple at a time ? .. the doc does not reveal that information and I haven't tried.